cmake/org.eclipse.cdt.cmake.core/src/org/eclipse/cdt/cmake/core/internal/CMakeToolChainManager.java

```diff
@@ -156,20 +156,18 @@ public void addToolChainFile(ICMakeToolChainFile file) {
 	public void removeToolChainFile(ICMakeToolChainFile file) {
 		init();
 		fireEvent(new CMakeToolChainEvent(CMakeToolChainEvent.REMOVED, file));
-		String tcId = makeToolChainId(file.getProperty(CMakeBuildConfiguration.TOOLCHAIN_TYPE),
-				file.getProperty(CMakeBuildConfiguration.TOOLCHAIN_ID));
-		filesByToolChain.remove(tcId);
-
-		String n = ((CMakeToolChainFile) file).n;
-		if (n != null) {
-			Preferences prefs = getPreferences();
-			Preferences tcNode = prefs.node(n);
-			try {
+		try {
+			String tcId = makeToolChainId(file.getToolChain());
+			filesByToolChain.remove(tcId);
+			String n = ((CMakeToolChainFile) file).n;
+			if (n != null) {
+				Preferences prefs = getPreferences();
+				Preferences tcNode = prefs.node(n);
 				tcNode.removeNode();
 				prefs.flush();
-			} catch (BackingStoreException e) {
-				Activator.log(e);
 			}
+		} catch (Exception e) {
```

Comment:
I would consider handling the CoreException, from org.eclipse.cdt.cmake.core.ICMakeToolChainFile.getToolChain(), separately. I think it's generally considered bad practice to catch the general Exception.

Reply:
Thanks for review @betamaxbandit updated the exception block for CoreException and BackingStoreException separately.